core/src/main/scala/slamdata/AutoBump.scala

```diff
@@ -0,0 +1,369 @@
+      .pullRequests
+      .draftPullRequest(
+        owner,
+        repoSlug,
+        NewPullRequestData(autoBumpCommitTitle(authorRepository),
+          s"This PR brought to you by sbt-trickle via $authorRepository. Changes:\n\n$changes"),
+        branchName,
+        "master")
+      .rethrow
+      .map(_.result)
+  }
+
+  def assignLabel(label: String, pullRequest: PullRequestDraft): IO[GHResponse[List[Label]]] = {
+    github
+      .issues
+      .addLabels(owner, repoSlug, pullRequest.number, List(label))
+  }
+
+  def markReady(pullRequest: PullRequestDraft): IO[Unit] = {
+    github
+      .pullRequests
+      .markReadyForReview(owner, repoSlug, pullRequest.node_id)
+      .rethrow
+      .ensure(new RuntimeException(s"Failed to mark pull request ${pullRequest.number} ready for review"))(_.result)
+      .void
+  }
+
+  def close(pullRequest: PullRequestDraft, title: String): IO[GHResponse[PullRequestDraft]] = {
+    val requestUpdate = PullRequestUpdate(title = Some(title), state = Some("closed"))
+    github.pullRequests.updatePullRequest(owner, repoSlug, pullRequest.number, requestUpdate)
+  }
+
+  def deleteBranch(pullRequest: PullRequestDraft): IO[Unit] = {
+    pullRequest.head.map { base =>
+      val branch = s"refs/heads/${base.ref}"
+      github.gitData.deleteReference(owner, repoSlug, branch).rethrow
+    }.sequence.void
+  }
+
+  def removeLabel(pullRequest: PullRequestDraft, label: String): IO[GHResult[List[Label]]] = {
+    github.issues.removeLabel(owner, repoSlug, pullRequest.number, label).rethrow
+  }
+
+  def createPullRequest(authorRepository: String, log: Logger): Boolean = {
+    val authenticated = s"https://_:$token@github.com/$owner/$repoSlug"
+
+    val tryUpdateDependencies: IO[Either[Warnings, (File, String, Option[PullRequestDraft], List[String], ChangeLabel)]] = {
+      for {
+        dir <- IO(Files.createTempDirectory("sbt-precog"))
+        dirFile = dir.toFile
+        _ <- run(s"git clone --depth 1 $authenticated ${dirFile.getPath}", log)
+        oldestPullRequest <- getOldestAutoBumpPullRequest
+        (flag, branchName) <- getBranch(oldestPullRequest)
+        _ <- run(s"git checkout $flag $branchName", log, workingDir = Some(dirFile))
+        lines <- run(s"$sbt trickleUpdateDependencies", log, workingDir = Some(dirFile))
+        changes = extractChanges(lines)
+        label = extractLabel(lines)
+        _ <- run(s"$sbt trickleIsUpToDate", log, workingDir = Some(dirFile))
+        updateResult <- run(s"$sbt update", log, merge = true, Some(dirFile)).attempt
+      } yield updateResult.bimap(_ => UpdateError, _ => (dirFile, branchName, oldestPullRequest, changes, label))
+    }
+
+    // TODO: add changes to commit message?
+    def tryCommit(dirFile: File): IO[Either[Warnings, Unit]] = {
+      for {
+        _ <- run(s"git add .", log, merge = true, workingDir = Some(dirFile))
+        result <- run(
+          Seq("git", "commit", "-m", autoBumpCommitTitle(authorRepository)),
```

Comment:
So what we *used* to do here is build a mini changelog from the origin repo(s). Basically, grab the origin repo, do a `git log vXXX..HEAD`, filter to the merge commits descending from each other (I can't remember exactly how this works, but git has a flag for it) and take the commit messages (which are derived from pull request titles by `sdmerge`).

I'd like to revive this at some point, but it's harder with trickle since you'd need to go clone *those* repos, and also the changelog thing wasn't really perfect anyway. This is fine for now, and we can refine it over time.

Reply:
It's even harder because, the way it was done, we knew for sure what the previous version was. It's trickier now, particularly with autobump PR updates. As for checking out... we don't need to do that. We can query github.

Reply:
OH, another thing... originally, every bump corresponded to a single repository version bump. Now we are combining bumps from multiple repositories.